File: 05-example-web-application/api-node/healthcheck/healthcheck.js
var http = require('http');

var options = {
  timeout: 2000,
  host: 'localhost',
  port: process.env.PORT || 3000,
  path: '/ping',
};

var request = http.request(options, (res) => {
  console.info('STATUS: ' + res.statusCode);
  process.exitCode = res.statusCode === 200 ? 0 : 1;
  process.exit();
});

request.on('error', function (err) {
  console.error('ERROR', err);
  process.exit(1);
});

request.end();

File: 05-example-web-application/api-golang/healthcheck/healthcheck.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"os"
	"time"
)

func main() {

	port, exists := os.LookupEnv("PORT")
	if !exists {
		port = "8080"
	}

	client := http.Client{
		Timeout: 2 * time.Second,
	}

	resp, err := client.Get("http://localhost:" + port + "/ping")
	if err != nil {
		log.Fatal(err)
	}

	// Print the HTTP Status Code and Status Name
	fmt.Println("HTTP Response Status:", resp.StatusCode, http.StatusText(resp.StatusCode))

	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
		fmt.Println("HTTP Status is in the 2xx range")
	} else {
		fmt.Println("Argh! Broken")
		os.Exit(1)
	}
}
